agora/Agora/Utils.hs

```diff
@@ -198,7 +201,32 @@ pfindInputWithStateThreadToken = plam $ \tokenSymbol inputs ->
             #== 1
       )
     # inputs
-    
+
+{- | Find an output containing exactly one token with the given currency symbol,
+    and with a PAddress that matches the given one.
+
+     @since 1.0.0
+-}
+pfindOutputWithStateThreadTokenAndAddress ::
+  forall tag.
+  ClosedTerm
+    ( PTagged tag PCurrencySymbol
+        :--> PAddress
+        :--> PBuiltinList PTxOut
+        :--> PMaybe PTxOut
+    )
+pfindOutputWithStateThreadTokenAndAddress = plam $ \tokenSymbol address outputs ->
+  pfind
+    # ( plam $ \output ->
+          ( ptaggedSymbolValueOf
+              # tokenSymbol
+              # (pfield @"value" # output)
+              #== 1
+          )
+            #&& (address #== (pfield @"address" # output))
```

Comment:
You're extracting 2 fields from `output`. It's more efficient to `pletFields` them

Reply:
Thanks, we got rid of the address check though as it's not needed, so only extracting the `value` now